File: DTOs/Objects/Professional.swift
import Foundation

/// Пространство имен `Professional` содержит типы данных для работы с информацией о клиентах.
///
/// В него входят параметры для запросов (`Parameters`) и модели ответов (`Responses`),
/// используемые для обмена данными между клиентом и сервером в контексте клиентских данных.
public enum Professional {
    public enum Parameters {}
    public enum Responses {}
}

//MARK: - Parameters -

public extension Professional.Parameters {
    
    struct Create: Parametable {
        
        public init() {}
    }
}

//MARK: - Responses -

public extension Professional.Responses {
    
    struct Partial: Responsable, Equatable {
        
        public var user: User.Responses.Partial
        
        public init(
            user: User.Responses.Partial
        ) {
            self.user = user
        }
    }
}

File: Android/shared/src/main/kotlin/shared/dto/objects/Professional.kt
@file:UseSerializers(URISerializer::class)

package shared.dto.objects

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import shared.dto.protocols.Parametable
import shared.dto.protocols.Responsable
import shared.serializers.URISerializer

object Professional {
    data object Parameters {
        @Serializable
        class Create : Parametable()
    }

    data object Responses {
        @Serializable
        data class Partial(
            val user: User.Responses.Partial
        ) : Responsable
    }
}
